packages/lib/gpc/src/gpcUtil.ts

```diff
@@ -192,16 +293,25 @@ export function resolvePODEntryIdentifier(
  * @param tuples a record mapping tuple names to tuple configurations
  * @returns a tuple of POD values if the entry is found and `undefined` otherwise
  */
-export function resolveTupleName(
-  tupleName: PODName,
+export function resolveTupleIdentifier(
+  tupleIdentifier: TupleIdentifier,
   pods: Record<PODName, POD>,
   tuples: Record<PODName, GPCProofTupleConfig>
 ): PODValueTuple | undefined {
+  const tupleName = tupleIdentifier.slice(7);
   const tupleEntries = tuples[tupleName].entries;
   const resolution = tupleEntries.map((entryId) =>
     resolvePODEntryIdentifier(entryId, pods)
   );
-  return resolution.some((value) => value === undefined)
+  resolution.forEach((value, i) => {
+    if (value === undefined) {
+      throw new ReferenceError(
+        `POD entry value identifier ${tupleEntries[i]} in tuple ${tupleName} does not have a value.`
+      );
+    }
+  });
+
+  return resolution.every((value) => value === undefined)
```

Comment:
I may be missing something, but isn't the `every` here redundant now?
The foreach above will throw if any value is undefined, so by the time you get here you know they're all defined and can just return the array.  Unless you intended to be checking `value.value` which shouldn't be undefined unless the type system is being violated (which can happen, but should be checked in the checking phase).